packages/backend-core/src/objectStore/objectStore.ts

```diff
@@ -524,3 +525,26 @@ export async function getReadStream(
   }
   return client.getObject(params).createReadStream()
 }
+
+/*
+Given a signed url like 'files/signed/tmp-files-attachments/app_123456/myfile.txt' extract
+the bucket and the path from it
+*/
+export function extractBucketAndPath(
+  url: string
+): { bucket: string; path: string } | null {
+  const baseUrl = url.split("?")[0]
+  if (!baseUrl.startsWith(signedFilePrefix)) {
+    return null
+  }
+
+  const parts = baseUrl.split("/")
+  if (parts.length < 5) {
+    return null
+  }
+
+  const bucket = parts[3]
```

Comment:
Same here. This deconstruction seems perfect for a regex. Could we use something like this? `^\/files\/signed\/(?'bucket'[^\/]+)\/(?'path'.+)$`

Reply:
Updated to regex!